vinyldns/endpoints.go

```diff
@@ -137,3 +92,26 @@ func batchRecordChangesEP(c *Client) string {
 func batchRecordChangeEP(c *Client, changeID string) string {
 	return concatStrs("", batchRecordChangesEP(c), "/", changeID)
 }
+
+func buildQuery(f ListFilter, nameFilterName string) string {
+	params := []string{}
+	query := "?"
+
+	if f.NameFilter != "" {
```

Comment:
This is more of an FYI than a correction needed, but I found out recently that the `python` requests library allows me to pass in a dictionary for the query params when making an HTTP request, and it builds the URL properly.  I wonder if go has something similar.  Maybe not, but I just did this and felt it was worth mentioning

Reply:
Thanks -- yes, I believe such a thing exists but I opted for the lofi DIY solution because...

1. I want to enforce the `ListFilter` type rather than accept an arbitrary map of values, which I don't _believe_ is ideal, as the Golang `net/url` package requires me to manually copy each struct field into a map, somewhat negating the value of using it.
2. I believe [go-querystring](https://github.com/google/go-querystring) offers the ability to convert structs to query parameters, though I don't know that its use dramatically reduces the code requires to accomplish this _and_ its use requires another external dependency.